pandas/tests/tools/test_to_datetime.py

```diff
@@ -2530,3 +2530,14 @@ def test_empty_string_datetime_coerce__unit():
     # verify that no exception is raised even when errors='raise' is set
     result = to_datetime([1, ""], unit="s", errors="raise")
     tm.assert_index_equal(expected, result)
+
+
+def test_to_datetime_monotonic_increasing_index():
+    # GH28238
+    times = date_range(datetime.now(), periods=1000, freq="h")
```

Comment:
Could you simplify `times` just to be a short Series of non-monotonic Timestamps? e.g. `Series([pd.Timestamp('2002'), pd.Timestamp('1992'), pd.Timestamp("2022")])`

Reply:
Updated